src/fn/sot457.ts

```diff
@@ -0,0 +1,231 @@
+import type { AnyCircuitElement, PcbSilkscreenPath } from "circuit-json"
+import { z } from "zod"
+import { rectpad } from "../helpers/rectpad"
+import { pillpad } from "src/helpers/pillpad"
+import { silkscreenRef, type SilkscreenRef } from "src/helpers/silkscreenRef"
+
+export const sot457_def = z.object({
+  fn: z.string().default("sot457"),
+  num_pins: z.literal(6).default(6),
+  h: z.string().default("2.5mm"),
+  w: z.string().default("2.7mm"),
+  wave: z.boolean().optional(),
+  reflow: z.boolean().optional(),
+  pillh: z.string().default("0.45mm"),
+  pillw: z.string().default("1.45mm"),
+  pl: z.string().default("0.8mm"),
+  pw: z.string().default("0.55mm"),
+  p: z.string().default("0.95mm"),
+})
+
+export const sot457_wave = z
+  .object({
+    fn: z.literal("sot457"),
+    num_pins: z.literal(6).default(6),
+    h: z.string().default("3mm"),
+    w: z.string().default("4mm"),
+    pillr: z.string().default("0.225mm"),
+    pillh: z.string().default("0.45mm"),
+    pillw: z.string().default("1.45mm"),
+    pl: z.string().default("1.45mm"),
+    pw: z.string().default("1.5mm"),
+    p: z.string().default("1.475mm"),
```

Comment:
just remove all of these parameters. they're not parameters because the user isn't allowed to change them, use constants instead